sdk/eventhub/event-hubs/test/stress/app/src/eventHubsStressTester.ts

```diff
@@ -0,0 +1,140 @@
+    const snapshotIntervalMs = !this.snapshotOptions.snapshotIntervalInMs
+      ? 5000
+      : this.snapshotOptions.snapshotIntervalInMs;
+
+    this.startedAt = new Date();
+    this.snapshotTimer = setInterval(this.snapshot.bind(this), snapshotIntervalMs);
+  }
+
+  public async init(options?: StressTestInitOptions) {
+    defaultClientAppInsights.commonProperties = {
+      // these will be reported with each event
+      testName: this.snapshotOptions.testName,
+    };
+
+    defaultClientAppInsights.trackEvent({
+      name: "start",
+      properties: {
+        ...options?.additionalEventProperties,
+        eventHubsName: options?.eventHubName,
+      },
+    });
+  }
+
+  public async snapshot(): Promise<void> {
+    const elapsedTimeInSeconds = (new Date().valueOf() - this.startedAt.valueOf()) / 1000;
+
+    this.eventProperties["elapsedTimeInSeconds"] = elapsedTimeInSeconds;
+    this.eventProperties["events.sent"] = this.eventsSentCount;
+    this.eventProperties["events.received"] = this.eventsReceivedCount;
+    this.eventProperties["errorCount"] = this._numErrors;
+    const { arrayBuffers, rss, heapUsed } = process.memoryUsage()
+    this.eventProperties["memory.arrayBuffers"] = arrayBuffers;
+    this.eventProperties["memory.rss"] = rss;
+    this.eventProperties["memory.heapUsed"] = heapUsed;
+    this._numErrors = 0;
+
+    defaultClientAppInsights.trackEvent({
+      name: "summary",
+      properties: this.eventProperties
+    });
+    defaultClientAppInsights.flush();
+
+    if (this.snapshotOptions.writeSnapshotInfoToConsole) {
+      console.log(JSON.stringify(this.eventProperties, undefined, 2));
+    }
+  }
+
+  public async endTest() {
+    console.log(`[BEGIN]: ending test...`);
+    await this.snapshot();
+    clearInterval(this.snapshotTimer);
+    // TODO: Log tracked messages in JSON
+    console.log(`[END]: ending test...`);
+  }
+}
+
+export function createEventHubsConsumerClient(options?: EventHubConsumerClientOptions): EventHubConsumerClient {
+  const consumerGroup = process.env.EVENTHUBS_CONSUMER_GROUP || "$Default";
+  const connectionString = process.env.EVENTHUBS_CONNECTION_STRING;
+  const hubName = process.env.EVENTHUB_NAME;
+  if (!connectionString || !consumerGroup || !hubName) {
+    throw new Error("EVENTHUBS_CONNECTION_STRING, EVENTHUB_NAME and EVENTHUBS_CONSUMER_GROUP have to be populated in the environment and are not!");
+  }
+  return new EventHubConsumerClient(consumerGroup, connectionString, hubName, options);
+}
+
+export function createEventHubsProducerClient(options?: EventHubClientOptions): EventHubProducerClient {
+  const eventHubName = process.env.EVENTHUB_NAME;
+  const connectionString = process.env.EVENTHUBS_CONNECTION_STRING;
+
+  if (!connectionString || !eventHubName) {
+    throw new Error("EVENTHUBS_CONNECTION_STRING and EVENTHUB_NAME have to be populated in the environment and are not!");
+  }
+
+  return new EventHubProducerClient(connectionString, eventHubName, options);
```

Comment:
Should we use buffered producer instead?

Reply:
it's just mainly used for the setup fo the tests.
If you want to particularly test the buffered producer, let's do that as a new issue/PR.

Reply:
Sorry I wasn't clear, I meant if there is no need for us to create the batches manually, we can just use the buffered producer to save a few lines of code 😊